pom.xml

```diff
@@ -267,6 +269,28 @@
   <dependencyManagement>
     <dependencies>
       <!-- dependencies are always listed in sorted order by groupId, artifactId -->
+      <dependency>
+        <groupId>com.amazonaws</groupId>
```

Comment:
So seems like the only class we use from aws-java-sdk-bundle is the SecretCache. Is this correct? if so, I am not very familiar with this Cache but how is this different from a map based cache ? 
Because of the usage of this class, we are having to declare this as a compile time dependency. I was wondering if we could make this a runtime dependency so we dont have to include this jar in our shipped binaries. Users using this feature need to a) configure a hive property to specify the SecretsManager class implementation(s) b) add the jars to the hive lib for runtime availability of the classes.

I am trying to avoid adding classloader footprint if possible.

Reply:
SecretCache is a class which understand how to fetch secrets from AWS Secrets Service and also provides caching support. The basic framework is pluggable, I can refactor the entire AWS secrets loading into another project which can be built when some profile is set.

Reply:
only this dependency adds ~200MB 
https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-bundle/1.12.331/